<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Introdu√ß√£o √° an√°lise de dados de sequenciamento NGS 16S rRNA</title>
    <meta charset="utf-8" />
    <meta name="author" content="Kelly Hidalgo" />
    <script src="index_files/header-attrs-2.5/header-attrs.js"></script>
    <link href="index_files/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="index_files/remark-css-0.0.1/rladies.css" rel="stylesheet" />
    <link href="index_files/remark-css-0.0.1/rladies-fonts.css" rel="stylesheet" />
    <script src="index_files/clipboard-2.0.6/clipboard.min.js"></script>
    <link href="index_files/shareon-1.4.1/shareon.min.css" rel="stylesheet" />
    <script src="index_files/shareon-1.4.1/shareon.min.js"></script>
    <link href="index_files/xaringanExtra-shareagain-0.2.4/shareagain.css" rel="stylesheet" />
    <script src="index_files/xaringanExtra-shareagain-0.2.4/shareagain.js"></script>
    <link href="index_files/xaringanExtra-clipboard-0.2.4/xaringanExtra-clipboard.css" rel="stylesheet" />
    <script src="index_files/xaringanExtra-clipboard-0.2.4/xaringanExtra-clipboard.js"></script>
    <script>window.xaringanExtraClipboard(null, {"button":"<i class=\"fa fa-clipboard\"><\/i>","success":"<i class=\"fa fa-check\" style=\"color: #90BE6D\"><\/i>","error":"<i class=\"fa fa-times-circle\" style=\"color: #F94144\"><\/i>"})</script>
    <link href="index_files/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
    <link href="index_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
    <link href="index_files/xaringanExtra-extra-styles-0.2.4/xaringanExtra-extra-styles.css" rel="stylesheet" />
    <link href="index_files/tile-view-0.2.4/tile-view.css" rel="stylesheet" />
    <script src="index_files/tile-view-0.2.4/tile-view.js"></script>
    <script src="index_files/kePrint-0.0.1/kePrint.js"></script>
    <link href="index_files/lightable-0.0.1/lightable.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Introdu√ß√£o √° an√°lise de dados de sequenciamento NGS 16S rRNA
## Usando R em an√°lises de microbiomas
### Kelly Hidalgo
### Grupo de Ecologia Microbiana e Multi-√¥micas
### 31/10/2020

---








&lt;style type="text/css"&gt;
.huge .remark-code { /*Change made here*/
  font-size: 150% !important;
}
.tiny .remark-code { /*Change made here*/
  font-size: 60% !important;
}
&lt;/style&gt;














<div>
<style type="text/css">.xaringan-extra-logo {
width: 110px;
height: 128px;
z-index: 0;
background-image: url(03.Images/GEMMHEX.png);
background-size: contain;
background-repeat: no-repeat;
position: absolute;
top:1em;right:1em;
}
</style>
<script>(function () {
  let tries = 0
  function addLogo () {
    if (typeof slideshow === 'undefined') {
      tries += 1
      if (tries < 10) {
        setTimeout(addLogo, 100)
      }
    } else {
      document.querySelectorAll('.remark-slide-content:not(.title-slide):not(.inverse):not(.hide_logo)')
        .forEach(function (slide) {
          const logo = document.createElement('div')
          logo.classList = 'xaringan-extra-logo'
          logo.href = null
          slide.appendChild(logo)
        })
    }
  }
  document.addEventListener('DOMContentLoaded', addLogo)
})()</script>
</div>



## Kelly Hidalgo

* üá®üá¥ Colombiana üá®üá¥

* Microbi√≥loga

* Mestre em gen√©tica e biologia molecular da Unicamp

* Doutoranda

* Grupo de Ecologia Microbiana e Multi-√¥micas da Divis√£o de Recursos Microbianos do CPQBA- UNICAMP

* √Årea de pesquisa: Microbiomas de √°reas contaminadas com petr√≥leo e derivados

&lt;img src="03.Images/ORCID.png" width="30%" style="display: block; margin: auto 0 auto auto;" /&gt;
---

## Introdu√ß√£o

R √© uma linguagem muito vers√°til, que √© amplamente usada em diferentes √°reas. Aqui voc√™ vai ter uma **pequena introdu√ß√£o** de como usar R na an√°lise de dados obtidos pelo sequenciamento massivo do gene 16S rRNA.
O sequenciamento massivo deste marcador filogen√©tico permite saber quais microrganismos est√£o presentes em um ambiente qualquer e √† abundancia relativa destes. 

&lt;img src="03.Images/workflow1.png" width="2769" /&gt;
---
class: center, middle

## Objetivo

Explorar a linguaguem R para an√°lises de **microbiomas** 

---

# An√°lise de Microbiomas

&lt;img src="03.Images/analises.png" width="100%" style="display: block; margin: auto;" /&gt;

---

## *Dataset* 

* Os dados fazem parte do tutorial ["Atacama soil microbiome"](https://docs.qiime2.org/2020.8/tutorials/atacama-soils/) de **Qiime2**. 

&lt;img src="03.Images/paper.png" width="80%" style="display: block; margin: auto;" /&gt;

&lt;font size="1"&gt; 
*Significant Impacts of Increasing Aridity on the Arid Soil Microbiome.* Julia W. Neilson, Katy Califf, Cesar Cardona, Audrey Copeland, Will van Treuren, Karen L. Josephson, Rob Knight, Jack A. Gilbert, Jay Quade, J. Gregory Caporaso, and Raina M. Maier. mSystems May 2017, 2 (3) e00195-16; DOI: 10.1128/mSystems.00195-16.
&lt;/font&gt;

* Deserto de Atacama no norte do üá®üá±. Esse deserto √© um dos ambientes mais √°ridos do planeta. Tem √°reas onde chove menos de um mm por d√©cada.

---

## *Dataset* 


* Os microbiomas a serem analisados neste tutorial foram coletados em dois transectos, **Baquedano** e **Yungay**, atrav√©s dos quais a umidade relativa m√©dia do solo est√° positivamente correlacionada com a eleva√ß√£o (eleva√ß√µes mais altas s√£o menos √°ridas, e, portanto, t√™m maior umidade relativa do solo)

* Ao longo desses transectos foram feitas covas em cada local e foram coletados solos em tr√™s profundidades em cada uma.


&lt;img src="03.Images/map.png" width="50%" style="display: block; margin: auto;" /&gt;

---

## Pre-requisitos

* [Download R Software](https://cran.r-project.org/)

* [Download R Studio](https://www.rstudio.com/products/rstudio/download/)

* [Dataset](https://docs.qiime2.org/2020.8/tutorials/atacama-soils/) 

    * `Arquivos.qza` (`table.qza`, `taxonomyclassification.qza` e `tree.qza`)
    * A tabela com os metadatos
  

**IMPORTANTE:** As an√°lises deste tutorial tamb√©m podem ser feitas apartir de tabelas em formato `tsv`, `txt`, `csv`. 

---

## Pacotes

.left-column[
### Principais

&lt;img src="03.Images/phyloseqsticker.png" width="80%" /&gt;

&lt;img src="03.Images/ggplot2.png" width="80%" /&gt;
]

.right.colum[
.center[
### Outros

qiime2R

vegan

gplots

ape

reshape2

gridExtra

broom

gburr
]
]

---

class: center, middle, inverse

.huge[
# M√£os a obra
]
---

## 0. Estabele√ßa seu diret√≥rio de trabalho


```r
setwd("caminho/para/seu/diretorio/de/trabalho")
```
### 0.1 Crie subdiret√≥rios
Com o intuito de que sua an√°lise fique organizada. Dentro de sua pasta de trabalho, crie subdiret√≥rios para cada tipo de informa√ß√£o. Por exemplo `00.Scripts/` para armazenar seus c√≥digos, `01.Data/`, para colocar todos os dados "input" da an√°lise, e por √∫ltimo `02.Results/` onde voc√™ pode guardar os "outputs" da an√°lise, como gr√°ficos, tabelas, etc. 


```r
## Cria a pasta para armazenar Scripts
*dir.create("00.Scripts/")
## Cria a pasta para guardar os dados de entrada
*dir.create("01.Data/")
## Cria a pasta para armazenar as an√°lises geradas
*dir.create("02.Results/")
```

---

## 1. Carregando a informa√ß√£o no ambiente R

### 1.1. Pacote qiime2R

Este pacote anexa os *output* do processamento das sequ√™ncias usando **QIIME2** (arquivos .qza)


```r
## Instala o pacote qiime2R que l√™ arquivos .qza (sa√≠da do QIIME2)
if (!requireNamespace("devtools", quietly = TRUE)){install.packages("devtools")}
devtools::install_github("jbisanz/qiime2R", force = TRUE)
```


```r
## Ativa a biblioteca do pacote qiime2R
library("qiime2R")
```


```r
## Confira a vers√£o do pacote
packageVersion("qiime2R")
```

```
[1] '0.99.35'
```

---
class: hide_logo

### 1.2. Lendo os artefatos do qiime2 (Frequency Table)

Para ler os arquivos de sa√≠da do **qiime2** (artefatos) vamos a usar a fun√ß√£o `read_qza()` do pacote "qiime2R"


```r
## Lendo o artefato table.qza, tabela com as freq√º√™ncias das ASVs (esp√©cies)
*ASVs &lt;- read_qza("01.Data/table.qza")
ASVs$type # imprime o tipo de artefacto
```

```
[1] "FeatureTable[Frequency]"
```

Para ver a informa√ß√£o inclu√≠da dentro do artefato, use o comando `names()`

```r
## Conte√∫do do arquivo table.qza
names(ASVs)
```

```
[1] "uuid"       "type"       "format"     "contents"   "version"   
[6] "data"       "provenance"
```

---

Para acessar aos dados de freq√º√™ncias das ASVs armazenados dentro do objeto, acesse os dados assim:

```r
## Mostra os primeiros 5 taxas e as primeiras 10 amostras
ASVs$data[1:5,1:10]
```

```
                                 BAQ2420.1.1 BAQ2420.1.2 BAQ2420.1.3 BAQ2420.2
1237d5925a7176fced9dda961a86c684           0           0          17        69
409faa5f5353e543bf6d99125c7c0e83           0           0           0         0
a7b877ae6d2f079a15b6b192a4425620           0          25          22        37
ef3fdbe1dcde754d91130cde6a4b4d61          39          20           0         0
96cbccca68ad868a78bb0604e4a41cf5           0           0           0         0
                                 BAQ2420.3 BAQ2462.1 BAQ2462.2 BAQ2462.3
1237d5925a7176fced9dda961a86c684        54        83        60         0
409faa5f5353e543bf6d99125c7c0e83         0         0         0         0
a7b877ae6d2f079a15b6b192a4425620        46        57         0        66
ef3fdbe1dcde754d91130cde6a4b4d61         0         0         0         0
96cbccca68ad868a78bb0604e4a41cf5         0         0         0         0
                                 BAQ2687.1 BAQ2687.2
1237d5925a7176fced9dda961a86c684         0         0
409faa5f5353e543bf6d99125c7c0e83         0         0
a7b877ae6d2f079a15b6b192a4425620         0         0
ef3fdbe1dcde754d91130cde6a4b4d61         0         0
96cbccca68ad868a78bb0604e4a41cf5         0         0
```
Cada linha representa um ASV (esp√©cie) e sua frequ√™ncia em cada amostra (colunas)
---

### 1.3. Lendo o Metadata

Se voc√™ quer usar o arquivo de metadata usado no qiime2, pode usar a fun√ß√£o `read_q2metada()` como se mostra embaixo. Se usa outro tipo de arquivo tsv ou csv, pode usar `read.table()`, `readr::read_tsv()`, ou `readr::csv()`.

.tiny[

```r
## L√™ o arquivo de metadata usado no qiime2
*metadata &lt;- read_q2metadata("01.Data/sample-metadata.tsv")
head(metadata) # Mostra as primeiras linhas do metadata
```

```
     SampleID   sample-id SampleName barcode-sequence elevation extract-concen
1 BAQ1370.1.2 BAQ1370.1.2    BAQ1370     GCCCAAGTTCAC      1370          0.019
2   BAQ1370.3   BAQ1370.3    BAQ1370     GCGCCGAATCTT      1370          0.124
3 BAQ1370.1.3 BAQ1370.1.3    BAQ1370     ATAAAGAGGAGG      1370          1.200
4 BAQ1552.1.1 BAQ1552.1.1    BAQ1552     ATCCCAGCATGC      1552          0.722
5   BAQ1552.2   BAQ1552.2    BAQ1552     GCTTCCAGACAA      1552          0.017
6 BAQ2420.1.1 BAQ2420.1.1    BAQ2420     ACACAGTCCTGA      2420          0.350
  amplicon-concentration extract-group-no transect-name site-name depth   ph
1                   0.95                B     Baquedano   BAQ1370     2 7.98
2                  17.46                E     Baquedano   BAQ1370     2   NA
3                   0.96                J     Baquedano   BAQ1370     3 8.13
4                  18.83                J     Baquedano   BAQ1552     1 7.87
5                   2.00                B     Baquedano   BAQ1552     2   NA
6                   7.40                H     Baquedano   BAQ2420     1 9.33
  toc    ec average-soil-relative-humidity relative-humidity-soil-high
1 525 6.080                          16.17                       23.97
2 771 6.080                          16.17                       23.97
3  NA    NA                          16.17                       23.97
4  NA    NA                          15.75                       35.36
5 223 1.839                          15.75                       35.36
6  NA    NA                          82.54                      100.00
  relative-humidity-soil-low percent-relative-humidity-soil-100
1                     11.420                               0.00
2                     11.420                               0.00
3                     11.420                               0.00
4                     11.100                               0.00
5                     11.100                               0.00
6                     41.135                              46.77
  average-soil-temperature temperature-soil-high temperature-soil-low
1                    22.61                35.210               12.460
2                    22.61                35.210               12.460
3                    22.61                35.210               12.460
4                    22.63                30.650               10.960
5                    22.63                30.650               10.960
6                    22.45                28.297               13.294
  vegetation percentcover
1         no            0
2         no            0
3         no            0
4         no            0
5         no            0
6         no            0
```
]

---
class: hide_logo

Perceba que tem diferentes vari√°veis, explicando seu dataset. Por exemplo, nome do transecto, eleva√ß√£o, humidade relativa do solo, vegeta√ß√£o, entre outras. Voc√™ pode colocar quantas vari√°veis voc√™ quiser, que v√£o te ajudar para que tua an√°lise fique mais robusta e interessante. 



&lt;table class=" lightable-classic lightable-hover table" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 9px; margin-left: auto; margin-right: auto;'&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; SampleID &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; transect-name &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; site-name &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; depth &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; ph &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; toc &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; ec &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; average-soil-relative-humidity &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; relative-humidity-soil-high &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; relative-humidity-soil-low &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; percent-relative-humidity-soil-100 &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; average-soil-temperature &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; temperature-soil-high &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370.1.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 7.98 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 525 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 6.080 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.17 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23.970 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.00 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.61 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.210 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 771 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 6.080 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.17 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23.970 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.00 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.61 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.210 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370.1.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ1370 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.13 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.17 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23.970 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.00 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.61 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.210 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ1552.1.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ1552 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 7.87 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 15.75 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.360 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.00 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.63 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 30.650 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ1552.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ1552 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 223 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.839 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 15.75 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.360 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.00 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.63 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 30.650 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 9.33 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.54 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41.135 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46.77 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 28.297 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 9.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 166 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.075 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.54 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41.135 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46.77 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 28.297 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 9.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 337 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.075 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.54 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41.135 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46.77 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 28.297 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 9.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 574 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.075 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.54 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41.135 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46.77 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 28.297 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.90 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; NA &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.54 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 41.135 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 46.77 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 22.45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 28.297 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 173 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.506 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 69.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 98.396 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 32.558 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 21.83 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 29.024 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 331 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.506 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 69.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 98.396 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 32.558 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 21.83 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 29.024 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 634 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.506 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 69.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 98.396 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 32.558 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 21.83 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 29.024 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 554 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.131 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 73.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 20.870 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 43.48 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 18.20 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 25.000 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 350 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.131 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 73.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 20.870 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 43.48 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 18.20 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 25.000 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 400 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.131 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 73.21 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 20.870 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 43.48 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 18.20 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 25.000 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 353 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.212 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 44.74 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.647 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.34 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 19.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 24.995 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.2 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 260 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.212 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 44.74 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.647 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.34 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 19.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 24.995 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.3 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 8.36 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 369 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.212 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 44.74 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16.647 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 11.34 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 19.08 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 24.995 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ3473.1 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Baquedano &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; BAQ3473 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 7.93 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3675 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.420 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 82.05 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100.000 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 42.090 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35.37 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 13.46 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 20.690 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---
class: hide_logo

### 1.4. Lendo os artefatos do qiime2 (Taxonomy Table)

Agora √© momento de importar o artefato que cont√©m a anota√ß√£o taxon√¥mica de cada um dos ASVs encontradas em nosso dataset.
.tiny[

```r
## L√™ o artefato da anota√ß√£o taxon√¥mica
*taxonomy &lt;- read_qza("01.Data/taxonomyclassification.qza")
head(taxonomy$data)[1:10,] # Mostra os dados armazenados no objeto taxonomy
```

```
                           Feature.ID
1    1237d5925a7176fced9dda961a86c684
2    409faa5f5353e543bf6d99125c7c0e83
3    a7b877ae6d2f079a15b6b192a4425620
4    ef3fdbe1dcde754d91130cde6a4b4d61
5    96cbccca68ad868a78bb0604e4a41cf5
6    a36b38f754f6abd278aeb9dbc7696343
NA                               &lt;NA&gt;
NA.1                             &lt;NA&gt;
NA.2                             &lt;NA&gt;
NA.3                             &lt;NA&gt;
                                                                                                                                                    Taxon
1                                                         d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae
2                    d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium
3    d__Bacteria; p__Actinobacteriota; c__Actinobacteria; o__Nitriliruptorales; f__Nitriliruptoraceae; g__Nitriliruptoraceae; s__uncultured_Nitriliruptor
4          d__Bacteria; p__Gemmatimonadota; c__Gemmatimonadetes; o__Gemmatimonadales; f__Gemmatimonadaceae; g__uncultured; s__uncultured_Gemmatimonadales
5                    d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium
6                 d__Bacteria; p__Actinobacteriota; c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteriaceae; g__Rubrobacter; s__uncultured_Rubrobacter
NA                                                                                                                                                   &lt;NA&gt;
NA.1                                                                                                                                                 &lt;NA&gt;
NA.2                                                                                                                                                 &lt;NA&gt;
NA.3                                                                                                                                                 &lt;NA&gt;
     Confidence
1     0.9587141
2     0.7912899
3     1.0000000
4     0.7549944
5     0.9967571
6     0.8145884
NA           NA
NA.1         NA
NA.2         NA
NA.3         NA
```
]
---
class: hide_logo

&lt;table class=" lightable-classic lightable-hover table" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; margin-left: auto; margin-right: auto; font-size: 9px; margin-left: auto; margin-right: auto;'&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Feature.ID &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Taxon &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; Confidence &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 1237d5925a7176fced9dda961a86c684 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9587141 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 409faa5f5353e543bf6d99125c7c0e83 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7912899 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; a7b877ae6d2f079a15b6b192a4425620 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Actinobacteria; o__Nitriliruptorales; f__Nitriliruptoraceae; g__Nitriliruptoraceae; s__uncultured_Nitriliruptor &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 1.0000000 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; ef3fdbe1dcde754d91130cde6a4b4d61 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Gemmatimonadota; c__Gemmatimonadetes; o__Gemmatimonadales; f__Gemmatimonadaceae; g__uncultured; s__uncultured_Gemmatimonadales &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7549944 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 96cbccca68ad868a78bb0604e4a41cf5 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9967571 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; a36b38f754f6abd278aeb9dbc7696343 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteriaceae; g__Rubrobacter; s__uncultured_Rubrobacter &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8145884 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 1694836fc379411d2b9aa087d68d571b &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9987904 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 11f7b172e09b77715d3bb2175a40b409 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9927076 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; ffd60d684f32e6fd5b47fe90095f9d34 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Thermoleophilia; o__Solirubrobacterales; f__67-14; g__67-14; s__uncultured_Rubrobacteria &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8791164 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; c434ee7f909f455fc2109ef1f741a2d4 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Actinobacteria; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Crossiella; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8738180 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; c6a6705874a49360a01c2b3fd39fe900 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Thermoleophilia; o__Solirubrobacterales; f__67-14; g__67-14 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9999977 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; f6c10a04d57159c0d64d6bc30c677471 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9986441 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 6b780e361cfc5f06def718518324bdcb &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Actinobacteria; o__Micrococcales; f__Micrococcaceae &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9999995 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; dc8a2f47b3d1dc2e1f5f805891976b29 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8990457 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 5987b97497dc2c6f7c3d690526ea34b0 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__Nitrosococcus; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8149896 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 5eee6273221f15d4e0cdc2c033f0dd72 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Thermoleophilia; o__Gaiellales; f__uncultured; g__uncultured; s__uncultured_actinobacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8078748 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 6e4d034e4b967b745639f825dc73ebe6 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Actinobacteria; o__0319-7L14; f__0319-7L14; g__0319-7L14; s__uncultured_actinobacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9338043 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; ddbc32632c9632d1dd746f28721f3a9f &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Firmicutes; c__Bacilli; o__Paenibacillales; f__Paenibacillaceae; g__Ammoniphilus &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9999836 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 28df1c1787c97637be4f8a7fe41c1105 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Acidimicrobiia; o__uncultured; f__uncultured; g__uncultured; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7224617 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 5c78314ff92e6fec9aa07acc1fa0dc24 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhizobiales; f__Xanthobacteraceae; g__Bradyrhizobium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8689161 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 352875c83f4adf659984d683666d2038 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Thermoleophilia; o__Gaiellales; f__Gaiellaceae; g__Gaiella; s__uncultured_Conexibacteraceae &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7948961 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; b65d67bf1cf90c3fcd9868edc2f35454 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__MB-A2-108; o__MB-A2-108; f__MB-A2-108; g__MB-A2-108; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9923323 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; f2e9c4d09183997ce7cdab5a93686b4c &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Gemmatimonadota; c__Gemmatimonadetes; o__Gemmatimonadales; f__Gemmatimonadaceae; g__uncultured; s__uncultured_Gemmatimonadales &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7589499 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; e043936a0f7ec9177eaaf79e4bbe8631 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nitrosococcales; f__Nitrosococcaceae; g__wb1-P19; s__uncultured_bacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9538545 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 4ca02e17ff1a389f9bdd9eb488a01563 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria; p__Actinobacteriota; c__Rubrobacteria; o__Rubrobacterales; f__Rubrobacteriaceae; g__Rubrobacter; s__uncultured_actinobacterium &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.7867505 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---


```r
## Quebra a coluna "taxon" em varias colunas organizadas com os n√≠veis taxon√¥micos
*taxonomy &lt;- parse_taxonomy(taxonomy$data)
tail(taxonomy) # Mostra as √∫ltimas 6 linhas
```

```
                                     Kingdom            Phylum
53dab22800179768292a4558ae312e7f d__Bacteria       Chloroflexi
0bbf87852ed648b187c1335890284295 d__Bacteria    Proteobacteria
299454bfc88f66af9967dec1eed4c7e2 d__Bacteria              &lt;NA&gt;
342ccdbaead927102589eceb0207d461 d__Bacteria    Proteobacteria
1d47e396375611a005274d179edcd824 d__Bacteria Verrucomicrobiota
7a18078f5b08fc24802a057c6aebaabf d__Bacteria       Chloroflexi
                                               Class              Order
53dab22800179768292a4558ae312e7f     Dehalococcoidia               S085
0bbf87852ed648b187c1335890284295 Gammaproteobacteria    Burkholderiales
299454bfc88f66af9967dec1eed4c7e2                &lt;NA&gt;               &lt;NA&gt;
342ccdbaead927102589eceb0207d461 Gammaproteobacteria Diplorickettsiales
1d47e396375611a005274d179edcd824    Verrucomicrobiae     Pedosphaerales
7a18078f5b08fc24802a057c6aebaabf              P2-11E             P2-11E
                                              Family           Genus
53dab22800179768292a4558ae312e7f                S085            S085
0bbf87852ed648b187c1335890284295    Burkholderiaceae       Ralstonia
299454bfc88f66af9967dec1eed4c7e2                &lt;NA&gt;            &lt;NA&gt;
342ccdbaead927102589eceb0207d461 Diplorickettsiaceae       Aquicella
1d47e396375611a005274d179edcd824     Pedosphaeraceae Pedosphaeraceae
7a18078f5b08fc24802a057c6aebaabf              P2-11E          P2-11E
                                              Species
53dab22800179768292a4558ae312e7f uncultured_bacterium
0bbf87852ed648b187c1335890284295                 &lt;NA&gt;
299454bfc88f66af9967dec1eed4c7e2                 &lt;NA&gt;
342ccdbaead927102589eceb0207d461 uncultured_bacterium
1d47e396375611a005274d179edcd824 uncultured_bacterium
7a18078f5b08fc24802a057c6aebaabf uncultured_bacterium
```

---
class: middle

### 1.5. Lendo os artefatos do qiime2 (Tree)
Agora vamos a ler e armazenar no ambiente do R, a √°rvore filogen√©tica.

```r
tree &lt;- read_qza("01.Data/tree.qza")
```

---
class: center, middle

###Estrutura Objeto Phyloseq


&lt;img src="03.Images/phyloseq.png" width="100%" style="display: block; margin: auto;" /&gt;

.tiny[
Instala√ß√£o:

```r
## Instala phyloseq
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("phyloseq")
```
]
---

## 2. Criando o objeto phyloseq
A fun√ß√£o `qza_to_phyloseq()` vincula varios artefatos lidos com `read_qza()` para criar um **objeto phyloseq** para as an√°lises subsequentes. 



```r
physeq&lt;-qza_to_phyloseq(
    features = "01.Data/table.qza",
    tree="01.Data/tree.qza",
    taxonomy="01.Data/taxonomyclassification.qza",
    metadata = "01.Data/sample-metadata.tsv"
    )
physeq #Mostra o objeto phyloseq criado
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1082 taxa and 54 samples ]
sample_data() Sample Data:       [ 54 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 1082 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1082 tips and 1076 internal nodes ]
```

Voc√™ tem um dataset com 54 amostras com 22 vari√°veis, e 1082 taxas diferentes em 7 n√≠veis taxon√¥micos. 

---
class: center, middle

### Fun√ß√µes

&lt;img src="03.Images/funcoes.png" width="100%" style="display: block; margin: auto;" /&gt;

---

## 2.1 Acessores do phyloseq

Dentro do phyloseq existem uma s√©rie de fun√ß√µes acessoras, que servem para voc√™ "acessar" na informa√ß√£o armazenada no objeto phyloseq criado. 

### nsamples()


```r
## Ativar a biblioteca do pacote phyloseq
library(phyloseq)
# imprime o n√∫mero de amostras do seu objeto
*nsamples(physeq)
```

```
[1] 54
```

---

### ntaxa()


```r
# imprime o n√∫mero de taxas do seu objeto
*ntaxa(physeq)
```

```
[1] 1082
```

### sample_names()


```r
 # imprime os nomes das amostras do seu objeto
*sample_names(physeq)
```

```
 [1] "BAQ2420.1.1" "BAQ2420.1.2" "BAQ2420.1.3" "BAQ2420.2"   "BAQ2420.3"  
 [6] "BAQ2462.1"   "BAQ2462.2"   "BAQ2462.3"   "BAQ2687.1"   "BAQ2687.2"  
[11] "BAQ2687.3"   "BAQ2838.1"   "BAQ2838.2"   "BAQ2838.3"   "BAQ3473.1"  
[16] "BAQ3473.2"   "BAQ3473.3"   "BAQ4166.1.1" "BAQ4166.1.2" "BAQ4166.1.3"
[21] "BAQ4166.2"   "BAQ4166.3"   "BAQ4697.1"   "BAQ4697.2"   "BAQ4697.3"  
[26] "YUN1005.1.1" "YUN1005.3"   "YUN1242.1"   "YUN1242.3"   "YUN1609.1"  
[31] "YUN2029.2"   "YUN3153.2"   "YUN3153.3"   "YUN3259.1.1" "YUN3259.1.2"
[36] "YUN3259.1.3" "YUN3259.2"   "YUN3259.3"   "YUN3346.1"   "YUN3346.2"  
[41] "YUN3346.3"   "YUN3428.1"   "YUN3428.2"   "YUN3428.3"   "YUN3533.1.1"
[46] "YUN3533.1.2" "YUN3533.1.3" "YUN3533.2"   "YUN3533.3"   "YUN3856.1.1"
[51] "YUN3856.1.2" "YUN3856.1.3" "YUN3856.2"   "YUN3856.3"  
```

---

### rank_names()

```r
 # imprime os n√≠veis taxon√¥micos do seu objeto
*rank_names(physeq)
```

```
[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"
```

### sample_variables()


```r
 # imprime as vari√°veis de sua data
*sample_variables(physeq)
```

```
 [1] "sample.id"                          "SampleName"                        
 [3] "barcode.sequence"                   "elevation"                         
 [5] "extract.concen"                     "amplicon.concentration"            
 [7] "extract.group.no"                   "transect.name"                     
 [9] "site.name"                          "depth"                             
[11] "ph"                                 "toc"                               
[13] "ec"                                 "average.soil.relative.humidity"    
[15] "relative.humidity.soil.high"        "relative.humidity.soil.low"        
[17] "percent.relative.humidity.soil.100" "average.soil.temperature"          
[19] "temperature.soil.high"              "temperature.soil.low"              
[21] "vegetation"                         "percentcover"                      
```

---

### sample_sums()

```r
# imprime a soma das freq√º√™ncias de todas as ASVs por amostra
*sample_sums(physeq)
```

```
BAQ2420.1.1 BAQ2420.1.2 BAQ2420.1.3   BAQ2420.2   BAQ2420.3   BAQ2462.1 
        733         831         881        1027         962        1397 
  BAQ2462.2   BAQ2462.3   BAQ2687.1   BAQ2687.2   BAQ2687.3   BAQ2838.1 
        866         728         944        1219        1037         481 
  BAQ2838.2   BAQ2838.3   BAQ3473.1   BAQ3473.2   BAQ3473.3 BAQ4166.1.1 
        438         342        1298         983        1900         942 
BAQ4166.1.2 BAQ4166.1.3   BAQ4166.2   BAQ4166.3   BAQ4697.1   BAQ4697.2 
       1559        1306        1305        1540        1782        1512 
  BAQ4697.3 YUN1005.1.1   YUN1005.3   YUN1242.1   YUN1242.3   YUN1609.1 
       2080        1978        1031        1470        1557        1384 
  YUN2029.2   YUN3153.2   YUN3153.3 YUN3259.1.1 YUN3259.1.2 YUN3259.1.3 
       1969         827        1112          28        1041         383 
  YUN3259.2   YUN3259.3   YUN3346.1   YUN3346.2   YUN3346.3   YUN3428.1 
        921        1352        1200         141        1125        1162 
  YUN3428.2   YUN3428.3 YUN3533.1.1 YUN3533.1.2 YUN3533.1.3   YUN3533.2 
       1914        1101        1363         920        1125        1987 
  YUN3533.3 YUN3856.1.1 YUN3856.1.2 YUN3856.1.3   YUN3856.2   YUN3856.3 
       1662         930        1624        1284        1249        1720 
```

---

### get_variable()

```r
# imprime os n√≠veis da vari√°vel desejada
*get_variable(physeq, varName = "elevation")
```

```
 [1] 2420 2420 2420 2420 2420 2462 2462 2462 2687 2687 2687 2838 2838 2838 3473
[16] 3473 3473 4166 4166 4166 4166 4166 4697 4697 4697 1005 1005 1242 1242 1609
[31] 2029 3153 3153 3259 3259 3259 3259 3259 3346 3346 3346 3428 3428 3428 3533
[46] 3533 3533 3533 3533 3856 3856 3856 3856 3856
```

### get_taxa()

```r
 # Imprime as freq√º√™ncias das ASVs na amostra desejada
*head(get_taxa(physeq, i = "BAQ3473.1"))
```

```
dfe7e74bb1e45ea73a95e1de0f813fb3 c9e32079e7c3e2a804a14ce56dfdcc05 
                               0                                0 
d8ffb44ef64f7589f69f8ade1a4f79d8 f48560eb73e52267345d8ff3e31e1791 
                               0                                0 
cf400229268468ae928415e072520baf 89827d1370a00626b70bd9cd9be6ab62 
                               0                                0 
```

---

## 2.2. Processadores do phyloseq (Filtering options)

Al√©m da fun√ß√µes acessoras, o **phyloseq** tem fun√ß√µes processadoras. 

### subset_taxa()

Por exemplo, voc√™ poderia estar interesad@ em ficar s√≥ com um grupo espec√≠fico de microrganismos, para isto o phyloseq tem a fun√ß√£o `subset_taxa()`.


```r
*Firmicutes_physeq &lt;- subset_taxa(physeq, Phylum == "Firmicutes")
## Confira
ntaxa(Firmicutes_physeq)
```

```
[1] 45
```

---


```r
head(tax_table(Firmicutes_physeq))[ , c("Class", "Order")] # Mostra a informa√ß√£o das colunas Phylum, Class e Order.
```

```
Taxonomy Table:     [6 taxa by 2 taxonomic ranks]:
                                 Class            
743b243a1305866076ee9822f873a166 "Bacilli"        
425bce5cd053e88ab6a4db6b7758e1d3 "Negativicutes"  
894f55d058e6e2ba106015899dd588e6 "Symbiobacteriia"
0760b09e3a3720666a9220d105f8ccba "Bacilli"        
e336fa16d75b13d2bad6b1d383e472ac "Bacilli"        
7452a179b9a24c364642eeaade4d266f "Bacilli"        
                                 Order                           
743b243a1305866076ee9822f873a166 "Erysipelotrichales"            
425bce5cd053e88ab6a4db6b7758e1d3 "Veillonellales-Selenomonadales"
894f55d058e6e2ba106015899dd588e6 "Symbiobacteriales"             
0760b09e3a3720666a9220d105f8ccba "Paenibacillales"               
e336fa16d75b13d2bad6b1d383e472ac "Paenibacillales"               
7452a179b9a24c364642eeaade4d266f "Paenibacillales"               
```
Agora voc√™ tem o objeto `Firmicutes_physeq`, o qual tem 45 taxas diferentes dentro, todas do phylum Firmicutes.

---
class: hide_logo

### subset_taxa()

E se voc√™ est√° interessado s√≥ na ordem "Bacilli" que pertence ao phylum "Firmicutes"? 


```r
*bacilli_physeq &lt;- subset_taxa(physeq, Phylum == "Firmicutes" &amp; Class == "Bacilli")
## Confira
ntaxa(bacilli_physeq)
```

```
[1] 43
```

```r
head(tax_table(bacilli_physeq)) [ , c("Phylum", "Class", "Order")]
```

```
Taxonomy Table:     [6 taxa by 3 taxonomic ranks]:
                                 Phylum       Class     Order               
743b243a1305866076ee9822f873a166 "Firmicutes" "Bacilli" "Erysipelotrichales"
0760b09e3a3720666a9220d105f8ccba "Firmicutes" "Bacilli" "Paenibacillales"   
e336fa16d75b13d2bad6b1d383e472ac "Firmicutes" "Bacilli" "Paenibacillales"   
7452a179b9a24c364642eeaade4d266f "Firmicutes" "Bacilli" "Paenibacillales"   
9279403ab31c9b2574b09cef10f08586 "Firmicutes" "Bacilli" "Paenibacillales"   
ddbc32632c9632d1dd746f28721f3a9f "Firmicutes" "Bacilli" "Paenibacillales"   
```

---
class: hide_logo

### tax_glom()

Outra fun√ß√£o processadora, √© `tax_glom()`, a qual serve para "colapsar" as ASVs num n√≠vel tax√¥nomico espec√≠fico. As vezes voc√™ quer fazer an√°lises s√≥ no n√≠vel de Clase. A fun√ß√£o `tax_glom()` aglomera as ASVs em um n√≠vel taxon√¥mico desejado.


```r
*class_physeq &lt;- tax_glom(physeq, "Class")
## Confira
ntaxa(class_physeq)
```

```
[1] 57
```

```r
head(tax_table(class_physeq))[ , c("Phylum", "Class")]
```

```
Taxonomy Table:     [6 taxa by 2 taxonomic ranks]:
                                 Phylum             Class            
cf400229268468ae928415e072520baf "Thermoplasmatota" "Thermoplasmata" 
c2252c3545754ab07a79ca051971f458 "Nanoarchaeota"    "Nanoarchaeia"   
a56b903521b8ab33a7878b35582803a8 "Crenarchaeota"    "Nitrososphaeria"
2dcce6012bdc7a9cb15900ee79beb2ed "Dependentiae"     "Babeliae"       
6aae0e15833d1c9e5eb9e1bdbcbfc5f9 "Patescibacteria"  "Saccharimonadia"
c3bd77a0f6486c74b4b713ce8a5f285b "Bacteroidota"     "Bacteroidia"    
```
Perceba que tem 57 Clases diferentes.

---
class: middle

### tax_glom()

Para entender melhor o que faz `tax_glom()`, imagine que sua informa√ß√£o √© assim:



&lt;table class=" lightable-classic lightable-hover table" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 9px; margin-left: auto; margin-right: auto;'&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; ASVs &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Kingdom &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Phylum &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Class &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Order &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Family &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Genus &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Abundance &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 743b243a1305866076ee9822f873a166 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridia &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridiales &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Peptococaceae &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Pelotomaculum &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 2 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 743b243a1305866076ee9822f873a166 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridia &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridiales &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Peptococaceae &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Pelotomaculum &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 3 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 894f55d058e6e2ba106015899dd588e6 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacilli &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillales &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillaceae &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillus &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 5 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 894f55d058e6e2ba106015899dd588e6 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacilli &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillales &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillaceae &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacillus &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 1 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 743b243a1305866076ee9822f873a166 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridia &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridiales &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Peptococaceae &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Pelotomaculum &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 7 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

Ap√≥s aplicar a fun√ß√£o `tax_glom()` ao n√≠vel de classe, ficaria assim:



&lt;table class=" lightable-classic lightable-hover table" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 9px; margin-left: auto; margin-right: auto;'&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; ASVs &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Kingdom &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Phylum &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Class &lt;/th&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt; Abundance &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 743b243a1305866076ee9822f873a166 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Clostridia &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 12 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; 894f55d058e6e2ba106015899dd588e6 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; d__Bacteria &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Firmicutes &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; Bacilli &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; 6 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---

### subset_samples()

Com a fun√ß√£o `subset_samples()` voc√™ pode separar amostras baseado nos metadatos. Por exemplo, imagine que voc√™ s√≥ quer fazer uma an√°lise separada das amostras do transecto "Baquedano", nos seguientes comandos vai criar um novo objeto s√≥ com essas amostras.


```r
*baquedano_physeq &lt;- subset_samples(physeq, transect.name %in% c("Baquedano"))
## Confira
head(sample_data(baquedano_physeq))[,1:5] # Mostra as primerias informa√ß√µes da nova tabela de metadatos
```

```
              sample.id SampleName barcode.sequence elevation extract.concen
BAQ2420.1.1 BAQ2420.1.1    BAQ2420     ACACAGTCCTGA      2420          0.350
BAQ2420.1.2 BAQ2420.1.2    BAQ2420     ATTATACGGCGC      2420          0.108
BAQ2420.1.3 BAQ2420.1.3    BAQ2420     ATTCAGATGGCA      2420          0.132
BAQ2420.2     BAQ2420.2    BAQ2420     TAAACGCGACTC      2420          0.089
BAQ2420.3     BAQ2420.3    BAQ2420     CCTCGGGTACTA      2420          0.083
BAQ2462.1     BAQ2462.1    BAQ2462     TTCACCTGTATC      2462          1.166
```

```r
baquedano_physeq # informa√ß√µes do novo objeto
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1082 taxa and 25 samples ]
sample_data() Sample Data:       [ 25 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 1082 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1082 tips and 1076 internal nodes ]
```
Voc√™ agora tem no novo objeto "baquedano_physeq" as amostras que correspondem ao transecto *Baquedano*.

---
class: hide_logo

### prune_samples()

Outro jeito de filtrar amostras √© com a fun√ß√£o `prune_samples()`. Com est√° fun√ß√£o voc√™ pode selecionar amostras baseado na posi√ß√£o delas nos metadatos, como se mostra a continua√ß√£o:

```r
samplesToKeep &lt;- sample_names(physeq) [1:15] # Cria um vetor, pegando as primeiras 15 amostras do objeto physeq (original) usando o acessor samples_names.
samplesToKeep ## Confira o vetor
```

```
 [1] "BAQ2420.1.1" "BAQ2420.1.2" "BAQ2420.1.3" "BAQ2420.2"   "BAQ2420.3"  
 [6] "BAQ2462.1"   "BAQ2462.2"   "BAQ2462.3"   "BAQ2687.1"   "BAQ2687.2"  
[11] "BAQ2687.3"   "BAQ2838.1"   "BAQ2838.2"   "BAQ2838.3"   "BAQ3473.1"  
```

```r
*first15samples_physeq &lt;- prune_samples(samplesToKeep, physeq)
## Confira o novo objeto
first15samples_physeq
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1082 taxa and 15 samples ]
sample_data() Sample Data:       [ 15 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 1082 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1082 tips and 1076 internal nodes ]
```

Voc√™ criou um novo objeto com as primeiras 15 amostras do dataset.

---

## 2.3. Manipula√ß√£o de abundancias

### rarefy_even_depth()

A fun√ß√£o `rarefy_even_depth()` normaliza todas as amostras √† uma mesma profundidade (soma das freq√º√™ncias de todas ASVs em cada amostra) e elimina as ASVs que desaparecem de todas as amostras como resultado da normaliza√ß√£o.

```r
*rare_physeq &lt;- rarefy_even_depth(physeq, rngseed = min(sample_sums(physeq)), replace = TRUE, trimOTUs = TRUE)
```

```
`set.seed(28)` was used to initialize repeatable random subsampling.
```

```
Please record this for your records so others can reproduce.
```

```
Try `set.seed(28); .Random.seed` for the full vector
```

```
...
```

```
579OTUs were removed because they are no longer 
present in any sample after random subsampling
```

```
...
```

---

Confira a soma das amostras no novo objeto que devem ser todas iguais


```r
sample_sums(rare_physeq) 
```

```
BAQ2420.1.1 BAQ2420.1.2 BAQ2420.1.3   BAQ2420.2   BAQ2420.3   BAQ2462.1 
         28          28          28          28          28          28 
  BAQ2462.2   BAQ2462.3   BAQ2687.1   BAQ2687.2   BAQ2687.3   BAQ2838.1 
         28          28          28          28          28          28 
  BAQ2838.2   BAQ2838.3   BAQ3473.1   BAQ3473.2   BAQ3473.3 BAQ4166.1.1 
         28          28          28          28          28          28 
BAQ4166.1.2 BAQ4166.1.3   BAQ4166.2   BAQ4166.3   BAQ4697.1   BAQ4697.2 
         28          28          28          28          28          28 
  BAQ4697.3 YUN1005.1.1   YUN1005.3   YUN1242.1   YUN1242.3   YUN1609.1 
         28          28          28          28          28          28 
  YUN2029.2   YUN3153.2   YUN3153.3 YUN3259.1.1 YUN3259.1.2 YUN3259.1.3 
         28          28          28          28          28          28 
  YUN3259.2   YUN3259.3   YUN3346.1   YUN3346.2   YUN3346.3   YUN3428.1 
         28          28          28          28          28          28 
  YUN3428.2   YUN3428.3 YUN3533.1.1 YUN3533.1.2 YUN3533.1.3   YUN3533.2 
         28          28          28          28          28          28 
  YUN3533.3 YUN3856.1.1 YUN3856.1.2 YUN3856.1.3   YUN3856.2   YUN3856.3 
         28          28          28          28          28          28 
```

Neste caso essa rarefa√ß√£o vai nos levar a perda de muita informa√ß√£o, pois tem uma amostras s√≥ com 28 sequ√™ncias. Voc√™ poderia eliminar amostras com baixa profundidade de sequenciamento para evitar a perdida de maior quantidade de informa√ß√£o. 

---

Agora que identificou que tem varias amostras com pouca profundidade, pode estabelecer um *threshold* para eliminar ditas amostras. 

Primeiro organize em ordem crescente as amostras pela soma total de ASVs 


```r
sort(sample_sums(physeq)) # Imprime e ordena de forma crescente as amostras segundo o tamanho
```

```
YUN3259.1.1   YUN3346.2   BAQ2838.3 YUN3259.1.3   BAQ2838.2   BAQ2838.1 
         28         141         342         383         438         481 
  BAQ2462.3 BAQ2420.1.1   YUN3153.2 BAQ2420.1.2   BAQ2462.2 BAQ2420.1.3 
        728         733         827         831         866         881 
YUN3533.1.2   YUN3259.2 YUN3856.1.1 BAQ4166.1.1   BAQ2687.1   BAQ2420.3 
        920         921         930         942         944         962 
  BAQ3473.2   BAQ2420.2   YUN1005.3   BAQ2687.3 YUN3259.1.2   YUN3428.3 
        983        1027        1031        1037        1041        1101 
  YUN3153.3   YUN3346.3 YUN3533.1.3   YUN3428.1   YUN3346.1   BAQ2687.2 
       1112        1125        1125        1162        1200        1219 
  YUN3856.2 YUN3856.1.3   BAQ3473.1   BAQ4166.2 BAQ4166.1.3   YUN3259.3 
       1249        1284        1298        1305        1306        1352 
YUN3533.1.1   YUN1609.1   BAQ2462.1   YUN1242.1   BAQ4697.2   BAQ4166.3 
       1363        1384        1397        1470        1512        1540 
  YUN1242.3 BAQ4166.1.2 YUN3856.1.2   YUN3533.3   YUN3856.3   BAQ4697.1 
       1557        1559        1624        1662        1720        1782 
  BAQ3473.3   YUN3428.2   YUN2029.2 YUN1005.1.1   YUN3533.2   BAQ4697.3 
       1900        1914        1969        1978        1987        2080 
```

---

### prune_samples()

Agora elimine as amostras com menor tamanho com a fun√ß√£o `prune_samples()`


```r
# Cria um novo objeto s√≥ com as amostras com tamanho maior de 200
*physeq2 &lt;- prune_samples(sample_sums(physeq)&gt;= 200, physeq)
sort(sample_sums(physeq2)) # confira
```

```
  BAQ2838.3 YUN3259.1.3   BAQ2838.2   BAQ2838.1   BAQ2462.3 BAQ2420.1.1 
        342         383         438         481         728         733 
  YUN3153.2 BAQ2420.1.2   BAQ2462.2 BAQ2420.1.3 YUN3533.1.2   YUN3259.2 
        827         831         866         881         920         921 
YUN3856.1.1 BAQ4166.1.1   BAQ2687.1   BAQ2420.3   BAQ3473.2   BAQ2420.2 
        930         942         944         962         983        1027 
  YUN1005.3   BAQ2687.3 YUN3259.1.2   YUN3428.3   YUN3153.3   YUN3346.3 
       1031        1037        1041        1101        1112        1125 
YUN3533.1.3   YUN3428.1   YUN3346.1   BAQ2687.2   YUN3856.2 YUN3856.1.3 
       1125        1162        1200        1219        1249        1284 
  BAQ3473.1   BAQ4166.2 BAQ4166.1.3   YUN3259.3 YUN3533.1.1   YUN1609.1 
       1298        1305        1306        1352        1363        1384 
  BAQ2462.1   YUN1242.1   BAQ4697.2   BAQ4166.3   YUN1242.3 BAQ4166.1.2 
       1397        1470        1512        1540        1557        1559 
YUN3856.1.2   YUN3533.3   YUN3856.3   BAQ4697.1   BAQ3473.3   YUN3428.2 
       1624        1662        1720        1782        1900        1914 
  YUN2029.2 YUN1005.1.1   YUN3533.2   BAQ4697.3 
       1969        1978        1987        2080 
```

---


```r
physeq2 # novo objeto
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1082 taxa and 52 samples ]
sample_data() Sample Data:       [ 52 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 1082 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 1082 tips and 1076 internal nodes ]
```

No novo objeto agora voc√™ tem 52 amostras que cumprem com o filtro usado.

### transform_sample_counts()

Outro jeito de normalizar √© transformando os valores das freq√º√™ncias das ASVs em valores de abundancias relativas ao tamanho total de cada amostra.


```r
count_to_prop &lt;- function(x) {return (x / sum (x))} # Cria uma fun√ß√£o que divide cada valor x (Freu√™ncias de ASVs) entre a soma total de cada amostra.
# aplica a fun√ß√£o no objeto original e cria um novo objeto com os valores transformados
*transf_physeq &lt;- transform_sample_counts(physeq, count_to_prop)
```

---


```r
sample_sums(transf_physeq) # Confira a soma das amostras do novo objeto, deve ser 1
```

```
BAQ2420.1.1 BAQ2420.1.2 BAQ2420.1.3   BAQ2420.2   BAQ2420.3   BAQ2462.1 
          1           1           1           1           1           1 
  BAQ2462.2   BAQ2462.3   BAQ2687.1   BAQ2687.2   BAQ2687.3   BAQ2838.1 
          1           1           1           1           1           1 
  BAQ2838.2   BAQ2838.3   BAQ3473.1   BAQ3473.2   BAQ3473.3 BAQ4166.1.1 
          1           1           1           1           1           1 
BAQ4166.1.2 BAQ4166.1.3   BAQ4166.2   BAQ4166.3   BAQ4697.1   BAQ4697.2 
          1           1           1           1           1           1 
  BAQ4697.3 YUN1005.1.1   YUN1005.3   YUN1242.1   YUN1242.3   YUN1609.1 
          1           1           1           1           1           1 
  YUN2029.2   YUN3153.2   YUN3153.3 YUN3259.1.1 YUN3259.1.2 YUN3259.1.3 
          1           1           1           1           1           1 
  YUN3259.2   YUN3259.3   YUN3346.1   YUN3346.2   YUN3346.3   YUN3428.1 
          1           1           1           1           1           1 
  YUN3428.2   YUN3428.3 YUN3533.1.1 YUN3533.1.2 YUN3533.1.3   YUN3533.2 
          1           1           1           1           1           1 
  YUN3533.3 YUN3856.1.1 YUN3856.1.2 YUN3856.1.3   YUN3856.2   YUN3856.3 
          1           1           1           1           1           1 
```

---

## 3. An√°lise explorat√≥ria
Agora vamos a fazer uma an√°lise explorat√≥ria de nossos dados. 

### 3.1 Diversidade Alfa 
A diversidade alfa √© igual a riqueza, ou seja n√∫mero de esp√©cies diferentes. Se refere √† diversidade dentro de cada amostra, e pode ser "quantificada" usando diferentes m√©tricas como, "Esp√©cies Observadas", o estimador "Chao1" e indices de diversidad como "Shannon" e "Simpson".

.tiny[
* Observed: √© a m√©trica mais b√°sica, √© simplesmente o n√∫mero de esp√©cies diferentes observadas.

* Chao1: √© um estimador, dando importancia para as esp√©cies ¬®raras¬®. 
$$
Observed + estimativa das esp√©cies n√£o observadas 
$$ 
* Shannon: Al√©m da riqueza contempla a uniformidade da distribui√ß√£o das abundancia das esp√©cies

* Simpson: √© igual a 1 - a probabilidade que duas bact√©rias escolhidas ao acaso na comunidade perten√ßam a esp√©cies diferentes. Tamb√©m √© o conhecido como indice de domin√¢ncia.
]
---
class: hide_logo

### plot_richness()

Com a fun√ß√£o `plot_richness()` do *Phyloseq* √© possivel graficar as m√©tricas desejadas.


```r
obj &lt;- physeq; p &lt;- plot_richness(obj, color = "SampleName", x = "SampleName", measures = c("Observed", "Chao1", "Shannon", "Simpson")); p
```

&lt;img src="index_files/figure-html/boxplot diversidade alfa-1.png" width="60%" style="display: block; margin: auto;" /&gt;

---
class: hide_logo, middle

Agora vamos informar para o R que o gr√°fico √© de tipo "boxplot" e adicionalmente vamos a customizar o gr√°fico usando fun√ß√µes do pacote ggplot2.

.pull-left[

```r
library(ggplot2) # Ativa a biblioteca do pacote ggplot2
p &lt;- p + geom_boxplot(aes(fill = "SampleName"), alpha=0.2, size=0.5) + theme_bw() # informa que quer que seja gr√°fico na forma de boxplot
p &lt;- p + theme(strip.background =element_rect(fill="black")) # personaliza a cor da faixa onde est√° o nome da cada m√©trica
p &lt;- p + theme(strip.text = element_text(colour = 'white', size = 10, face = "bold")) # personaliza a letra na mesma faixa anterior, sendo de cor branca, tamnaho 10 e negrita
p &lt;- p + theme(plot.margin = unit(c(0,1,0,1), "cm")) # Coloca margens
p &lt;- p + theme(axis.text.x = element_blank()) # personaliza o texto no eixo x
p &lt;- p + theme(axis.text.y = element_text(angle = 0, hjust = 1, size=8,color="black")) # personaliza o texto no eixo y
p &lt;- p + labs(x = "SampleName", y = "Alpha Diversity Measure") # Coloca nome nos eixos
p 
```
]

.pull-right[
![](index_files/figure-html/plot-label-out-1.png)&lt;!-- --&gt;
]

Com ggplot2 voc√™ pode personalizar todos os elementos que compoem o gr√°fico. 

---
class: middle

### *Ready-to-print*

`ggsave()`


```r
ggsave("02.Results/alpha_diversity_boxplot.tiff", units = "in", width=14, height=8, dpi=600, compression = 'lzw', p) # Salva o gr√°fico na pasta de resultados em formato .tiff com as medidas desejadas e a resolu√ß√£o de 600 dpi.
ggsave("02.Results/alpha_diversity_boxplot.pdf", units="in", width=11, height=8, dpi=600, p) # Salva o gr√°fico em formato pdf
```

---
class: hide_logo, middle

Outro jeito de graficar sem ser boxplot √©:

.pull-left[

```r
diversity_indexes &lt;- c("Observed", "Chao1", "Shannon", "Simpson") # Cria um vector com as m√©tricas de diversidade alfa que v√£o ser usadas.
p &lt;- plot_richness(physeq, color = "elevation", x = "SampleName", shape = "vegetation", measures = diversity_indexes) + # plota os valores das m√©tricas de diversidade, usando uma escala de cor para a eleva√ß√£o, e formas para diferenciar entre com ou sem vegeta√ß√£o
ggtitle("Alpha diversity index") + # coloca t√≠tulo
scale_alpha_discrete(name="elevation") + # escala de cor pra vari√°vel discreta
geom_point(size=2) # personaliza o tamnho do ponto
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out2-1.png)&lt;!-- --&gt;
]

---
class: middle

E se voc√™ simplesmente quer guardar todos os dados das m√©tricas de diversidade alfa em uma tabela?


```r
alpha_diversity_data &lt;- estimate_richness(physeq, measures = diversity_indexes)
```

```r
write.csv(alpha_diversity_data, "02.Results/alpha_diversity.csv")
```

&lt;table class=" lightable-classic lightable-hover table" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 9px; margin-left: auto; margin-right: auto;'&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;font-weight: bold;"&gt;   &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; Observed &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; Chao1 &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; se.chao1 &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; Shannon &lt;/th&gt;
   &lt;th style="text-align:right;font-weight: bold;"&gt; Simpson &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 26 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 26 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.910587 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9311451 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 31 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 31 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.138456 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9456109 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.1.3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.148694 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9352106 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 38 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 38 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.477656 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9651143 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2420.3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 38 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 38 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.451550 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9642658 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 35 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.181441 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9401084 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.734510 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9160617 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2462.3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 33 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 33 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.119220 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9396284 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 31 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 31 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.176438 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9497495 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 27 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 27 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.972910 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9367150 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2687.3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 39 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 39 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.371845 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9554562 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 23 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.869501 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9307532 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.2 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 19 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 19 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.624246 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9118242 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ2838.3 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 16 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 2.484743 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.8879484 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; BAQ3473.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 45 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 3.292579 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 0.9354833 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

---
class: hide_logo

### ANOVA

Voc√™ pode fazer ANOVA para testar se uma determinada m√©trica √© significativamente diferente em fun√ß√£o de algumas vari√°veis de seu estudo. Por exemplo, o transeto tem um efeito significativo no n√∫mero de esp√©cies observadas.


```r
data &lt;- cbind(sample_data(physeq), alpha_diversity_data) # Junta o componente sample_data do objeto physeq com a tabela das m√©tricas de diversidade criada anteriormente
data[1:15, c(2,4,23,24,26,27)] # Observe
```

```
            SampleName elevation Observed Chao1  Shannon   Simpson
BAQ2420.1.1    BAQ2420      2420       26    26 2.910587 0.9311451
BAQ2420.1.2    BAQ2420      2420       31    31 3.138455 0.9456109
BAQ2420.1.3    BAQ2420      2420       35    35 3.148694 0.9352106
BAQ2420.2      BAQ2420      2420       38    38 3.477656 0.9651143
BAQ2420.3      BAQ2420      2420       38    38 3.451550 0.9642658
BAQ2462.1      BAQ2462      2462       35    35 3.181441 0.9401084
BAQ2462.2      BAQ2462      2462       23    23 2.734510 0.9160617
BAQ2462.3      BAQ2462      2462       33    33 3.119220 0.9396284
BAQ2687.1      BAQ2687      2687       31    31 3.176437 0.9497495
BAQ2687.2      BAQ2687      2687       27    27 2.972910 0.9367150
BAQ2687.3      BAQ2687      2687       39    39 3.371845 0.9554562
BAQ2838.1      BAQ2838      2838       23    23 2.869501 0.9307532
BAQ2838.2      BAQ2838      2838       19    19 2.624246 0.9118242
BAQ2838.3      BAQ2838      2838       16    16 2.484743 0.8879484
BAQ3473.1      BAQ3473      3473       45    45 3.292579 0.9354833
```

---
class: hide_logo

### ANOVA


```r
physeq.anova &lt;- aov(Observed ~ transect.name, data) # Aplica o modelo de ANOVA para compara√ß√£o entre a m√©trica de esp√©cies observadas (Observed) e a vari√°vel "transect.name"
summary(physeq.anova) # Imprime o resumo da ANOVA
```

```
              Df Sum Sq Mean Sq F value Pr(&gt;F)
transect.name  1     42   42.44   0.293  0.591
Residuals     52   7537  144.95               
```

Ao observar o *valor de p* da estat√≠stica F usada no modelo de an√°lises de variancias, pode se concluir que o transeto n√£o tem efeito no n√∫mero de esp√©cies observadas, ou seja n√£o existe diferencias estat√≠sticamente significativas entre o n√∫mero de esp√©cies observadas nas amostras dos dois transetos. 
Mas ser√° que a eleva√ß√£o tem efeito no indice de diversidade de Shannon?


```r
physeq.anova2 &lt;- aov(Shannon ~ elevation, data) # Aplica o modelo de ANOVA para compara√ß√£o entre a m√©trica de esp√©cies observadas (Observed) e a vari√°vel "transect.name"
summary(physeq.anova2) # Imprime o resumo da ANOVA
```

```
            Df Sum Sq Mean Sq F value  Pr(&gt;F)   
elevation    1  2.235  2.2350   8.655 0.00486 **
Residuals   52 13.429  0.2582                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```

Sendo um valor de p baixo (n√≠vel de confian√ßa 95%), pode se concluir que a **eleva√ß√£o tem um efeito significativo no indice de diversidade Shannon**.

---

### 3.1 Diversidade Beta

A diversidade Beta responde √† pergunta qu√£o similares ou diferentes s√£o duas comunidades (amostras)? Para "medir" essas diferen√ßas se usam matrizes de distancias. As m√©tricas de distancias mais usadas para microbiomas s√£o:
* Jaccard: qualitativa (ausencia/presencia)

* Bray-Curtis: quantitativa, al√©m da presente/n√£o presente leva em conta as abundancias

* Unweighted Unifrac: Distancias filogen√©ticas. Qualitativa.

* Weighted Unifrac: Distancias filogen√©ticas. Quantitativa.

Um jeito de visualizar as matrizes de similaridades √© usando heatmap. O Phyloseq tem a fun√ß√£o **extra** `plot_dist_as_heatmap()` que permite graficar as distancias.

---
class: middle

### Instalandos as fun√ß√µes extra do Phyloseq


```r
# Para carregar as fun√ß√µes extras √© necess√°rio ativar os seguintes pacotes. Sen√£o tiver eles instaldos use o comando install.packages() e depois ative as bibliotecas
library(reshape2) 
library(ggplot2)
library(phyloseq)
library(ape)
library(gridExtra)
# Cria um vetor com o`nome dos scripts das fun√ß√µes extras
scripts &lt;- c("graphical_methods.R", 
             "tree_methods.R", 
             "plot_merged_trees.R", 
             "specificity_methods.R", 
             "ternary_plot.R", 
             "richness.R", 
             "edgePCA.R", 
             "copy_number_correction.R", 
             "import_frogs.R", 
             "prevalence.R", 
             "compute_niche.R")

urls &lt;- paste0("https://raw.githubusercontent.com/mahendra-mariadassou/phyloseq-extended/master/R/", scripts) # Concatena o vetor contido no reposit√≥rio no github com o criado com os nomes dos scripts.

for (url in urls) {
  source(url) 
} # "instala" cada nova fun√ß√£o
```

---

### plot_distances()

Para gerar as matrizes de distancias use a fun√ß√£o `distance()`


```r
distanceMethodList ## Lista os m√©todos dispon√≠veis
```

```
$UniFrac
[1] "unifrac"  "wunifrac"

$DPCoA
[1] "dpcoa"

$JSD
[1] "jsd"

$vegdist
 [1] "manhattan"  "euclidean"  "canberra"   "bray"       "kulczynski"
 [6] "jaccard"    "gower"      "altGower"   "morisita"   "horn"      
[11] "mountford"  "raup"       "binomial"   "chao"       "cao"       

$betadiver
 [1] "w"   "-1"  "c"   "wb"  "r"   "I"   "e"   "t"   "me"  "j"   "sor" "m"  
[13] "-2"  "co"  "cc"  "g"   "-3"  "l"   "19"  "hk"  "rlb" "sim" "gl"  "z"  

$dist
[1] "maximum"   "binary"    "minkowski"

$designdist
[1] "ANY"
```

---

### plot_distance()

```r
metodos_distancias &lt;- unlist(distanceMethodList) # cria um vetor com os m√©todos dispon√≠veis
```

Gere as matrizes de distancias


```r
dist.bc &lt;- distance(physeq, method = "bray") # Gera a matriz de dist√¢ncia usando a m√©trica Bray-Curtis
dist.jaccard &lt;- distance(physeq, method = "jaccard") # Matriz usando Jaccard
dist.unifrac &lt;- distance(physeq, method = "unifrac") # Matriz usando Unweighted Unifrac
dist.wunifrac &lt;- distance(physeq, method = "wunifrac") # Matriz usando Weighted Unifrac
```

Ap√≥s criadas as matrizes de distancias, agora pode gerar os heatmaps


```r
p1 &lt;- plot_dist_as_heatmap(dist.bc, title = "Bray-Curtis") + 
  theme(axis.text.x = element_blank()) + 
  theme(axis.text.y = element_blank())# Heatmap matriz de distancia Bray-Curtis
p2 &lt;- plot_dist_as_heatmap(dist.jaccard, title = "Jaccard") + 
  theme(axis.text.x = element_blank()) + 
  theme(axis.text.y = element_blank()) # Heatmap matriz de distancia Jaccard
p3 &lt;- plot_dist_as_heatmap(dist.unifrac, title = "Unifrac") + 
  theme(axis.text.x = element_blank()) + 
  theme(axis.text.y = element_blank()) # Heatmap matriz de distancia Unifrac
p4 &lt;- plot_dist_as_heatmap(dist.wunifrac, title = "Weighted Unifrac") + 
  theme(axis.text.x = element_blank()) + 
  theme(axis.text.y = element_blank()) # Heatmap matriz de distancia Weighted Unifrac
```

---
class: middle


```r
pf &lt;- grid.arrange(p1,p2,p3,p4, ncol = 2, nrow = 2); pf
ggsave("02.Results/matrizesdistancia.tiff", units = "in", width=14, height=8, dpi=600, compression = 'lzw', pf) # Salva o gr√°fico na pasta de resultados em formato .tiff com as medidas desejadas e a resolu√ß√£o de 600 dpi.
```

---
class: middle

### Algumas interpreta√ß√µes:

  * Jaccard maior que Unifrac: os taxa s√£o distintos mas filogen√©ticamente relacionados.
  * Unifrac maior que Weighted Unifrac: Taxa abundantes em ambas comunidades s√£o filogen√©ticamente pr√≥ximos

Um outro jeito mais usado para visualizar essas diferencias calculadas nas matrizes √© usando gr√°ficos de ordina√ß√£o como PCA, PCoA, NMDS...

### ordinate()

A ordina√ß√£o √© feita atrav√©s da fun√ß√£o `ordinate()`.



```r
ord.bc &lt;- ordinate(physeq, method = "MDS", distance = "bray") # Pode usar diretamente o nome da m√©trica
ord.bc &lt;- ordinate(physeq, method = "MDS", distance = dist.bc) # ou usar as matrizes geradas anteriormente
ord.jaccard &lt;- ordinate(physeq, method = "MDS", distance = "jaccard")
ord.unifrac &lt;- ordinate(physeq, method = "MDS", distance = "unifrac")
ord.wunifrac &lt;- ordinate(physeq, method = "MDS", distance = "wunifrac")
```

---
class: hide_logo

### plot_ordination()

Uma vez feitas as ordina√ß√µes, agora com a fun√ß√£o `plot_ordination()` grafique

.pull-left[

```r
p1 &lt;- plot_ordination(physeq, ord.bc, color = "site.name") + 
  theme_bw()
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out3-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_samples()

A fun√ß√£o `plot_samples()` √© similar √† `plot_ordination()` mas pode replicar um parametro a mais, veja s√≥:

.pull-left[

```r
p1 &lt;- plot_samples(physeq, ord.bc, 
                   color = "site.name", 
                   replicate = "site.name", 
                   shape = "vegetation") +
  theme_bw()
p1
```
]
.pull-right[
![](index_files/figure-html/plot-label-out4-1.png)&lt;!-- --&gt;
]

---
class: hide_logo

### plot_samples()

Perceba que as cores n√£o facilitam a diferencia√ß√£o entre uma amostra e outra. Neste ["Site"](http://medialab.github.io/iwanthue/) voc√™ pode criar paletas de cores, √© s√≥ colocar o n√∫mero de cores que voc√™ precisa, copiar e colar os c√≥digos "HEX json" e guardar eles no R em forma de vetor, assim:

.pull-left[

```r
# Crie uma paleta de 17 cores
colors17 &lt;- c("#8b53a0", "#5bab4b", 
              "#c954bf", "#99a142", 
              "#7463d6", "#d89733", 
              "#5372ba", "#d14e36", 
              "#4facd8", "#de4379", 
              "#4aaa86", "#a24576", 
              "#8e6229", "#a495e1", 
              "#dc946c", "#da87bf", 
              "#c05e66") 
p1 &lt;- p1 + scale_color_manual(values = c(colors17))
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out5-1.png)&lt;!-- --&gt;
]
---

### Personalizando o gr√°fico

.pull-left[

```r
# Personalizando o gr√°fico
p1 &lt;- p1 + geom_point(size=2.0) + 
  theme(plot.title = element_text(hjust = 0.0, 
                                  size = 16, colour =  "black", 
                                  face= "bold")) + ggtitle("Bray-Curtis Distance") + 
  scale_shape_manual(values = c(15,17)) + 
  theme(legend.position = "none") + 
  theme(axis.title.x = element_text(angle = 0, hjust = 0.5, 
                                    size = 12, color = "black", face = "bold")) +  
  theme(axis.title.y = element_text(angle = 90, hjust = 0.5, 
                                    size = 12, color = "black", face = "bold"))
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out6-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

Voc√™ pode graficar todas as m√©tricas e junta-las em um mesmo gr√°fico com varios p√°neis, como √© mostrado a continua√ß√£o usando as fun√ß√µes `plot_ordination()` de **phyloseq** e `ggarrange()` do pacote **ggpubr**

### Bray-Curtis

.pull-left[

```r
# Bray Curtis
p1 &lt;- plot_ordination(physeq, ord.bc, 
                      color = "site.name", shape = "vegetation") +
    theme_bw() + 
  scale_color_manual(values = c(colors17)) + 
  geom_point(size=5, alpha = 0.75) + 
  theme(plot.title = element_text(hjust = 0.0, size = 16, colour =  "black", face= "bold")) + ggtitle("Bray-Curtis Distance") + 
  scale_shape_manual(values = c(15,17)) + 
  theme(legend.position = c(0.85, 0.85), 
        legend.justification = c(0, 1), legend.box.just= 'right')  + 
  theme(axis.title.x = element_text(angle = 0, hjust = 0.5, 
                                    size = 12, color = "black", face = "bold")) +  theme(axis.title.y = element_text(angle = 90, 
                                                                                                                     hjust = 0.5, size = 12, color = "black", face = "bold"))
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out7-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### Jaccard

.pull-left[

```r
# Jaccard
p2 &lt;- plot_ordination(physeq, ord.jaccard, 
                      color = "site.name", shape = "vegetation") +
    theme_bw() + 
  scale_color_manual(values = c(colors17)) + 
  geom_point(size=5, alpha = 0.75) + 
  theme(plot.title = element_text(hjust = 0.0, 
                                  size = 16, colour =  "black", face= "bold")) + ggtitle("Jaccard Distance") + scale_shape_manual(values = c(15,17)) + 
  theme(legend.position = "none", 
        legend.justification = c(0, 1), 
        legend.box.just = 'right')  + 
  theme(axis.title.x = element_text(angle = 0, hjust = 0.5, 
                                    size = 12, color = "black", face = "bold")) +  theme(axis.title.y = element_text(angle = 90, 
                                                                                                                     hjust = 0.5, size = 12, color = "black", face = "bold"))
p2
```
]

.pull-right[
![](index_files/figure-html/plot-label-out8-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### Unifrac

.pull-left[

```r
# Unifrac
p3 &lt;- plot_ordination(physeq, ord.unifrac, 
                      color = "site.name", shape = "vegetation") +
  theme_bw() + scale_color_manual(values = c(colors17)) + 
  geom_point(size=5, alpha = 0.75) + 
  theme(plot.title = element_text(hjust = 0.0, size = 16, 
                                  colour =  "black", face= "bold")) + ggtitle("Unifrac Distance") + scale_shape_manual(values = c(15,17)) + 
  theme(legend.position = "none") + 
  theme(axis.title.x = element_text(angle = 0, 
                                    hjust = 0.5, size = 12, color = "black", face = "bold")) +  theme(axis.title.y = element_text(angle = 90, 
                                                                                                                                  hjust = 0.5, size = 12, color = "black", face = "bold"))
p3
```
]

.pull-right[
![](index_files/figure-html/plot-label-out9-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### Weigthed Unifrac

.pull-left[

```r
# Weighted Unifrac
p4 &lt;- plot_ordination(physeq, ord.wunifrac, 
                      color = "site.name", shape = "vegetation") +
  theme_bw() + scale_color_manual(values = c(colors17)) + 
  geom_point(size=5, alpha = 0.75) + 
  theme(plot.title = element_text(hjust = 0.0, size = 16, colour =  "black", face= "bold")) + ggtitle("Weighted Unifrac Distance") + 
  scale_shape_manual(values = c(15,17)) + 
  theme(legend.position = "none") + 
  theme(axis.title.x = element_text(angle = 0, 
                                    hjust = 0.5, size = 12, color = "black", face = "bold")) +  theme(axis.title.y = element_text(angle = 90, 
                                                                                                                                  hjust = 0.5, size = 12, color = "black", face = "bold"))
p4
```
]

.pull-right[
![](index_files/figure-html/plot-label-out10-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### ggarrange()

Junte os gr√°ficos usando a fun√ß√£o `ggarrange()`

.pull-left[

```r
library(ggpubr) 
pf &lt;- ggarrange(p1, p2, p3, p4, ncol = 2, 
                nrow = 2, common.legend = TRUE, 
                legend = "right"); pf
```
]

.pull-right[
![](index_files/figure-html/plot-label-out11-1.png)&lt;!-- --&gt;
]
---

### Loops sempre facilitam a vida!

Observe o uso dos loops para graficar todas os m√©todos de distancia usados anteriormente.  Primeiro ser√£o escolhidos os m√©todos mais comuns para a an√°lise de microbiomas (Bray-Curtis, Jaccard, Unifrac e Weighted Unifrac).


```r
metodos_distancias &lt;- metodos_distancias[c(1,2,8,10)] # Escolha os m√©todos desejados dentro do vetor
```

A continua√ß√£o ser√° feito um loop para que cada m√©todo de distancia escolhido, seja plotado e armazenado numa lista

```r
plots_list &lt;- vector("list", length(metodos_distancias))
names(metodos_distancias) = metodos_distancias
for( i in metodos_distancias) {
  # Gere as matrizes de distancias
  dist &lt;- distance(physeq, method = i)
  # Calcule a ordina√ß√£o
  ord &lt;- ordinate(physeq, "MDS", distance = dist)
  # Grafique
  p &lt;- NULL
  p &lt;- plot_samples(physeq, ord, color = "site.name", shape = "vegetation") + scale_color_manual(values = c(colors17))
  # Add t√≠tulo
  p &lt;- p + ggtitle(paste("MDS usando o m√©todo de distancia ", i, sep = ""))
  # Salve o gr√°fico 
  plots_list[[i]] = p
}
```

---

Depois criou uma tabela com os calculos de ordina√ß√£o de cada m√©trica em cada amostra


```r
library(plyr)
df &lt;- ldply(plots_list, function(x) x$data) # divide a lista, aplica a fun√ß√£o e retorna em um data.frame
names(df)[1] &lt;- "distancia" # Colocou nome na coluna do m√©todo de distancia
(df)[1:10,c(1:5,7,11,12)] # Observe
```

```
   distancia       Axis.1      Axis.2   sample.id SampleName elevation
1    unifrac -0.028180859 -0.18132478 BAQ2420.1.1    BAQ2420      2420
2    unifrac -0.055280255 -0.19082916 BAQ2420.1.2    BAQ2420      2420
3    unifrac -0.016132143 -0.18819542 BAQ2420.1.3    BAQ2420      2420
4    unifrac -0.129919438  0.06025319   BAQ2420.2    BAQ2420      2420
5    unifrac -0.168140553  0.08022548   BAQ2420.3    BAQ2420      2420
6    unifrac -0.268079624  0.16394189   BAQ2462.1    BAQ2462      2462
7    unifrac -0.054524087  0.15289973   BAQ2462.2    BAQ2462      2462
8    unifrac  0.005771959 -0.04322313   BAQ2462.3    BAQ2462      2462
9    unifrac -0.055423815 -0.23038862   BAQ2687.1    BAQ2687      2687
10   unifrac -0.297360196  0.15285964   BAQ2687.2    BAQ2687      2687
   transect.name site.name
1      Baquedano   BAQ2420
2      Baquedano   BAQ2420
3      Baquedano   BAQ2420
4      Baquedano   BAQ2420
5      Baquedano   BAQ2420
6      Baquedano   BAQ2462
7      Baquedano   BAQ2462
8      Baquedano   BAQ2462
9      Baquedano   BAQ2687
10     Baquedano   BAQ2687
```

---
class: hide_logo

### Graficando

.pull-left[

```r
p &lt;- ggplot(df, aes(Axis.1, Axis.2, color = site.name, shape = vegetation)) + 
  geom_point(size= 2, alpha = 0.5) + 
  facet_wrap(~distancia, scales = "free") + 
  ggtitle("MDS usando varias m√©tricas de distancia") + theme_bw() + 
  scale_color_manual(values = c(colors17)) + 
  theme(plot.title = element_text(hjust = 0.0, 
                                  size = 16, colour =  "black", face= "bold")) +
  scale_shape_manual(values = c(15,17)) + 
  theme(axis.title.x = element_text(angle = 0, 
                                    hjust = 0.5, size = 12, color = "black", face = "bold")) +
  theme(axis.title.y = element_text(angle = 90, 
                                    hjust = 0.5, size = 12, color = "black", face = "bold"))
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out12-1.png)&lt;!-- --&gt;
]
Perceba que nesse gr√°fico conjunto, voc√™ n√£o consigue ver as porcentagens dos eixos de cada gr√°fico. Voc√™ pode "imprimir" cada gr√°fico por separado

---

### Bray-Curtis


```r
print(plots_list[["bray"]]) # Imprime o gr√°fico usando Bray-Curtis
```

&lt;img src="index_files/figure-html/distance7-1.png" width="65%" style="display: block; margin: auto;" /&gt;

---

### Jaccard


```r
print(plots_list[["jaccard"]]) # Imprime o gr√°fico usando Jaccard
```

&lt;img src="index_files/figure-html/distance8-1.png" style="display: block; margin: auto;" /&gt;

---
class: hide_logo

### Outros jeitos de graficar as dist√¢ncias!

Usando poligonos, isto facilita comparar as dist√¢ncias entre amostras.

.pull-left[

```r
p1 &lt;- plot_ordination(physeq, ord.bc, 
                      type = "SampleID", color = "site.name", shape = "vegetation") + geom_polygon(aes(fill=site.name)) + 
  geom_point(size = 3) + 
  ggtitle("Bray-Curtis Distance") + 
  scale_color_manual(values = colors17) + 
  scale_fill_manual(values = colors17)
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out13-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### Split Graphic

Vamos criar um gr√°fico de tipo split com a ordina√ß√£o das amostras e dos 50 ASVs mais abundantes, representados no n√≠vel de phylum

Usando a fun√ß√£o `prune_taxa()` voc√™ pode eliminar ASVs n√£o desejadas. Al√©m aplicando a fun√ß√£o `taxa_sums()` em conjunto com `sort()`vamos a selecionar os 50 ASVs mais abundantes do objeto orginal


```r
prune_physeq &lt;- prune_taxa(names(sort(taxa_sums(physeq), TRUE))[1:50], physeq) 
prune_physeq # imprime as informa√ß√£oes do novo objeto
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 50 taxa and 54 samples ]
sample_data() Sample Data:       [ 54 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 50 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 50 tips and 49 internal nodes ]
```

Observe que agora tem um objeto com as 50 esp√©cies mais abundantes

---
class: hide_logo

### Graficando

.pull-left[

```r
p1 &lt;- plot_ordination(prune_physeq, ord.bc, 
                      type = "split", 
                      color = "Phylum", 
                      shape = "vegetation", 
                      label = "site.name", 
                      title = "Bray-Curtis: Samples/Phylum");
p1
```
]

.pull-right[
![](index_files/figure-html/plot-label-out15-1.png)&lt;!-- --&gt;
]

Salve o gr√°fico usando `ggsave()`

---

### 4. Curvas de rarefa√ß√£o

As curvas de rarefa√ß√£o ou satura√ß√£o permitem observar se de fato foi amostrada toda a diversidade real das amostras. No eixo x encontra-se o n√∫mero de sequ√™ncias e no eixo y o n√∫mero de esp√©cies observadas. Assim, quando a curva atinge o *plateau* indica que n√£o importa quantas sequ√™ncias mais voc√™ tiver n√£o vai encontrar mais esp√©cies diferentes.

.pull-left[

```r
# Precisa das fun√ß√µes extras do pacote Phyloseq
p &lt;- ggrare(physeq, step = 100, color = "sample.id", se = FALSE,)
```
]

.pull-right[

```
rarefying sample BAQ2420.1.1
rarefying sample BAQ2420.1.2
rarefying sample BAQ2420.1.3
rarefying sample BAQ2420.2
rarefying sample BAQ2420.3
rarefying sample BAQ2462.1
rarefying sample BAQ2462.2
rarefying sample BAQ2462.3
rarefying sample BAQ2687.1
rarefying sample BAQ2687.2
rarefying sample BAQ2687.3
rarefying sample BAQ2838.1
rarefying sample BAQ2838.2
rarefying sample BAQ2838.3
rarefying sample BAQ3473.1
rarefying sample BAQ3473.2
rarefying sample BAQ3473.3
rarefying sample BAQ4166.1.1
rarefying sample BAQ4166.1.2
rarefying sample BAQ4166.1.3
rarefying sample BAQ4166.2
rarefying sample BAQ4166.3
rarefying sample BAQ4697.1
rarefying sample BAQ4697.2
rarefying sample BAQ4697.3
rarefying sample YUN1005.1.1
rarefying sample YUN1005.3
rarefying sample YUN1242.1
rarefying sample YUN1242.3
rarefying sample YUN1609.1
rarefying sample YUN2029.2
rarefying sample YUN3153.2
rarefying sample YUN3153.3
rarefying sample YUN3259.1.1
rarefying sample YUN3259.1.2
rarefying sample YUN3259.1.3
rarefying sample YUN3259.2
rarefying sample YUN3259.3
rarefying sample YUN3346.1
rarefying sample YUN3346.2
rarefying sample YUN3346.3
rarefying sample YUN3428.1
rarefying sample YUN3428.2
rarefying sample YUN3428.3
rarefying sample YUN3533.1.1
rarefying sample YUN3533.1.2
rarefying sample YUN3533.1.3
rarefying sample YUN3533.2
rarefying sample YUN3533.3
rarefying sample YUN3856.1.1
rarefying sample YUN3856.1.2
rarefying sample YUN3856.1.3
rarefying sample YUN3856.2
rarefying sample YUN3856.3
```

![](index_files/figure-html/plot-label-out16-1.png)&lt;!-- --&gt;
]

---
class: middle

### Curvas de rarefa√ß√£o


```r
p &lt;- p + facet_wrap(~site.name) + # Cria pan√©is baseado no sitio de coleta
  theme_bw() + theme(legend.position = "none") + xlab("Sequ√™ncias anotadas") +
  ylab("Esp√©cies Observadas") + ggtitle("Curvas de rarefa√ß√£o");p
```

&lt;img src="index_files/figure-html/rarefaction curves2-1.png" width="65%" height="65%" style="display: block; margin: auto;" /&gt;

---

### 5. √Årvore filogen√©tica

### plot_tree()

Usando a fun√ß√£o `plot_tree()` √© poss√≠vel graficar a √°rvore filogen√©tica do objeto phyloseq.


```r
plot_tree(physeq)
```

&lt;img src="index_files/figure-html/tree1-1.png" width="50%" height="50%" style="display: block; margin: auto;" /&gt;

---
class: hide_logo

### plot_tree()

Mas √© claro que vai ser uma √°rvore imposs√≠vel de interpretar, tem 1082 esp√©cies diferentes. Por√©m, usando fun√ß√µes de filtragem e processadoras para focar em uma parte dos dados, plotar a √°rvore poderia ser muito √∫til. Agora use o objeto com os 50 ASVs mais abundantes criado anteriormente 

.pull-left[

```r
plot_tree(prune_physeq)
```
]

.pull-right[
![](index_files/figure-html/plot-label-out17-1.png)&lt;!-- --&gt;
]

Por default √° fun√ß√£o grafica a √°rvore preta. Os pontos do lado de cada ASV (bra√ßo), √© cada amostra no que dito ASV foi observado. 

---
class: hide_logo

### plot_tree()

Agora vamos a por cores nessa √°rvore. Vamos a incluir de novo os pontos com cada amostra, s√≥ que vamos a diferenciar cada uma por diferentes cores.

.pull-left[

```r
plot_tree(prune_physeq, 
          nodelabf = nodeplotdefault(), 
          ladderize = "left", 
          color = "site.name") + 
  scale_color_manual(values = c(colors17))
# O argumento nodelabf = nodeplodefault() coloca o bootstraping nos n√≥dulos
```
]

.pull-right[
![](index_files/figure-html/plot-label-out18-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_tree()

Talvez seja interessante pra voc√™ colocar cores segundo a classe ou qualquer outro n√≠vel taxon√≥mico

.pull-left[

```r
plot_tree(prune_physeq, 
          nodelabf = nodeplotdefault(), 
          ladderize = "left", 
          color = "Class") + 
  scale_color_viridis_d()
# O argumento nodelabf = nodeplodefault() coloca o bootstraping nos n√≥dulos
```
]

.pull-right[
![](index_files/figure-html/plot-label-out19-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_tree()

Tamb√©m √© poss√≠vel usar shapes para mapear outra vari√°vel quando esta tem 6 ou menos n√≠veis. E pode ser usado ao mesmo tempo que cor para mapear uma vari√°vel diferente.

.pull-left[

```r
plot_tree(prune_physeq, 
          nodelabf = nodeplotdefault(), 
          ladderize = "left", 
          color = "site.name", 
          shape = "Phylum", 
          label.tips = "Order") 
# O argumento nodelabf = nodeplodefault() coloca o bootstraping nos n√≥dulos
```
]

.pull-right[
![](index_files/figure-html/plot-label-out20-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_tree()

Agora imagine que voc√™ quer plotar a √°rvore do objeto que criou l√° no come√ßo com as ASVs do phylum Firmicutes e que al√©m quer colocar informa√ß√£o de abundancia. 

.pull-left[

```r
p &lt;- plot_tree(Firmicutes_physeq, 
               nodelabf=nodeplotblank, 
               color="site.name",
               label.tips="Genus", 
               size="abundance", 
               plot.margin=0.6, 
               base.spacing=0.05, # D√° mais espa√ßo em cada tip para ficar mais arrumado
          min.abundance=25)  + # coloca a abundancia em n√∫meros em cada esp√©cie que tem pelo menos 25 de abundancia 
scale_color_manual(values = colors17)
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out21-1.png)&lt;!-- --&gt;
]
---

### 6. Composi√ß√£o da comunidade

### plot_bar()

O gr√°fico mais simples para apresentar a composi√ß√£o da comunidade √© o Bar Plot com a fun√ß√£o `plot_bar()`.


```r
plot_bar(physeq)
```

&lt;img src="index_files/figure-html/barplot1-1.png" width="60%" height="60%" style="display: block; margin: auto;" /&gt;

Horr√≠vel n√©? Como sempre, √© necess√°rio usar mais fun√ß√µes de **phyloseq** para filtrar e processar os dados e de **ggplot** para personalizar o gr√°fico.

---
class: hide_logo

### plot_bar()

√â muito mais informativo graficar as abundancias relativas, que as freq√º√™ncias, ent√£o use o objeto `transf_physeq`, que voc√™ criou anteriormente, onde transformou os valores para abundancias relativas.

Diferencie as clases com diferentes cores.

```r
plot_bar(transf_physeq, fill = "Class")
```

&lt;img src="index_files/figure-html/barplot2-1.png" width="60%" height="60%" style="display: block; margin: auto;" /&gt;

---
class: hide_logo

### plot_bar()

Voc√™ pode criar pan√©is separados para cada transecto, para facilitar a interpreta√ß√£o

.pull-left[

```r
plot_bar(transf_physeq, fill = "Class") + 
  facet_wrap(~transect.name, 
             scales = "free_x", nrow = 1)
```
]

.pull-right[
![](index_files/figure-html/plot-label-out22-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_bar()

Ainda o gr√°fico est√° muito polu√≠do e √© dif√≠cil tirar conclus√µes. Filtre os 15 ASVs mais abundantes usando a fun√ß√£o `prune_species()`.

.pull-left[

```r
TopASVss &lt;- names(sort(taxa_sums(transf_physeq), TRUE)[1:15]); 
first15ASVs   &lt;- prune_taxa(TopASVss, transf_physeq)

plot_bar(first15ASVs, fill = "Class") + 
  facet_wrap(~transect.name, 
             scales = "free_x", nrow = 1) + 
  geom_bar(aes(color=Class, fill=Class), 
           stat="identity", position="stack")
```
]

.pull-right[
![](index_files/figure-html/plot-label-out23-1.png)&lt;!-- --&gt;
]

As barras n√£o chegam at√© 1.0 de abundancia!

---

### Barplot2


```r
# Use a fun√ß√£o filter_taxa() para selecionar as ASVs observadas mais de 1 vezes
physeq3 &lt;- filter_taxa(physeq, function(x) mean(x) &gt; 1, TRUE)
physeq3 
```

```
phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 252 taxa and 54 samples ]
sample_data() Sample Data:       [ 54 samples by 22 sample variables ]
tax_table()   Taxonomy Table:    [ 252 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 252 tips and 249 internal nodes ]
```

No novo objeto agora tem 252 taxas diferentes. A seguir colapse as ASVs no n√≠vel de Phylum e transforme as freq√º√™ncias em abundancias relativas


```r
phylum_physeq3 &lt;- tax_glom(physeq3, "Phylum")
phylum_physeq4 &lt;- transform_sample_counts(phylum_physeq3, function(x) x / sum(x) )
head(sample_sums(phylum_physeq4)) # confira a soma 
```

```
BAQ2420.1.1 BAQ2420.1.2 BAQ2420.1.3   BAQ2420.2   BAQ2420.3   BAQ2462.1 
          1           1           1           1           1           1 
```

---
class: hide_logo

### Continuando...

Agora ser√£o feitas uma s√©rie de transforma√ß√µes nos dados para graficar as barras com uma nova categoria "*outros*"

.pull-left[

```r
data &lt;- psmelt(phylum_physeq4) # Transforme os dodos em data.frame
data$Phylum &lt;- as.character(data$Phylum) # Converta os phylum em caracter
data$Phylum[data$Abundance &lt; 0.01] &lt;- "&lt; 1% abund."# Cria e nomea um grupo de phyla com abund√£ncia menor que 1% 
# Plote a nova data transformada 
p &lt;- ggplot(data=data, aes(x=sample.id, y=Abundance, fill=Phylum))
p &lt;- p + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = colors17) + 
  theme(axis.text.x = element_text(angle = 90)) +
  theme(legend.position="right") + 
  guides(fill=guide_legend(nrow=17)) +
  ggtitle("Phylum Relative Abundance") + 
  facet_wrap(~site.name, scales = "free_x", nrow = 1)
p &lt;- p + theme(strip.background =element_rect(fill="black"))
p &lt;- p + theme(strip.text = element_text(colour = 'white', size = 7, face = "bold", angle = 90))
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out24-1.png)&lt;!-- --&gt;
]

---
class: hide_logo

### plot_heatmap

Outro jeito de graficar a composi√ß√£o da comunidade √© usando heatmaps com a fun√ß√£o `plot_heatmap()`. Tradicionalmente, os heatmap t√™m sido usados para enfatizar os dados que est√£o acima ou abaixo de um limite como cores ‚Äúquentes‚Äù ou ‚Äúfrias‚Äù, respectivamente.

.pull-left[

```r
Gammaproteobacteria_physeq &lt;- subset_taxa(physeq, 
                                          Phylum=="Proteobacteria" &amp; 
                                            Class == "Gammaproteobacteria")
# Filtre s√≥ ASVs que pertecem ao phylum Proteobacteria
plot_heatmap(Gammaproteobacteria_physeq)
```
]

.pull-right[
![](index_files/figure-html/plot-label-out26-1.png)&lt;!-- --&gt;
]
---
class: hide_logo

### plot_heatmap()
A fun√ß√£o `plot_heatmap()` tem varios argumentos para fazer modifica√ß√µes no gr√°fico com o intuito de fazer ele mais informativo e explicativo.

.pull-left[

```r
p &lt;- plot_heatmap(Gammaproteobacteria_physeq, "NMDS", "bray", "site.name", "Family")
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out27-1.png)&lt;!-- --&gt;
]

---
class: hide_logo

### plot_heatmap()

Usando pan√©is

.pull-left[

```r
p &lt;- plot_heatmap(Gammaproteobacteria_physeq, "NMDS", "bray", "site.name", "Family") + 
  facet_wrap(~site.name, scales = "free_x", nrow = 1) + 
  theme(strip.background =element_rect(fill="black")) + 
  theme(strip.text = element_text(colour = 'white', size = 5, face = "bold", angle = 90))
p
```
]

.pull-right[
![](index_files/figure-html/plot-label-out28-1.png)&lt;!-- --&gt;
]
---

### Outra fun√ß√£o: heatmap.2()

Finalmente, com a fun√ß√£o `heatmap.2()` do pacote **gplots** √© poss√≠vel graficar um heatmap com clusteriza√ß√£o.


```r
library(gplots) # fun√ß√£o heatmap.2()
library(vegan) # Calcula as dist√¢ncias
library(RColorBrewer) # Paletas de cores

condition &lt;- function(x) {x &gt; 5} # Cria uma condi√ß√£o, s√≥ valores maiores que 5
temp_1 &lt;- genefilter_sample(first15samples_physeq, condition, 4) # aplica a condi√ß√£o usando a fun√ß√£o genefilter_sample()
obj_taxon_cond &lt;- prune_taxa(temp_1, first15samples_physeq) # Elimina as ASVs que n√£o cumprem com a condi√ßao
tax_table &lt;-  (tax_table(obj_taxon_cond)) # Extrai a tax_table do objeto filtrado 
# Extrai cada coluna/amostra da otu_table e calcula a abundancia relativa
t1 &lt;- round(((otu_table(obj_taxon_cond))[,1] / (colSums(otu_table(first15samples_physeq))[1]))*100, digits = 3)
t2 &lt;- round(((otu_table(obj_taxon_cond))[,2] / (colSums(otu_table(first15samples_physeq))[2]))*100, digits = 3)
t3 &lt;- round(((otu_table(obj_taxon_cond))[,3] / (colSums(otu_table(first15samples_physeq))[3]))*100, digits = 3)
t4 &lt;- round(((otu_table(obj_taxon_cond))[,4] / (colSums(otu_table(first15samples_physeq))[4]))*100, digits = 3)
t5 &lt;- round(((otu_table(obj_taxon_cond))[,5] / (colSums(otu_table(first15samples_physeq))[5]))*100, digits = 3)
t6 &lt;- round(((otu_table(obj_taxon_cond))[,6] / (colSums(otu_table(first15samples_physeq))[6]))*100, digits = 3)
t7 &lt;- round(((otu_table(obj_taxon_cond))[,7] / (colSums(otu_table(first15samples_physeq))[7]))*100, digits = 3)
t8 &lt;- round(((otu_table(obj_taxon_cond))[,8] / (colSums(otu_table(first15samples_physeq))[8]))*100, digits = 3)
t9 &lt;- round(((otu_table(obj_taxon_cond))[,9] / (colSums(otu_table(first15samples_physeq))[9]))*100, digits = 3)
t10 &lt;- round(((otu_table(obj_taxon_cond))[,10] / (colSums(otu_table(first15samples_physeq))[10]))*100, digits = 3)
t11 &lt;- round(((otu_table(obj_taxon_cond))[,11] / (colSums(otu_table(first15samples_physeq))[11]))*100, digits = 3)
t12 &lt;- round(((otu_table(obj_taxon_cond))[,12] / (colSums(otu_table(first15samples_physeq))[12]))*100, digits = 3)
t13 &lt;- round(((otu_table(obj_taxon_cond))[,13] / (colSums(otu_table(first15samples_physeq))[13]))*100, digits = 3)
t14 &lt;- round(((otu_table(obj_taxon_cond))[,14] / (colSums(otu_table(first15samples_physeq))[14]))*100, digits = 3)
t15 &lt;- round(((otu_table(obj_taxon_cond))[,15] / (colSums(otu_table(first15samples_physeq))[15]))*100, digits = 3)
```

---

### Continua... üôÑ


```r
## Junta cada coluna/amostra em uma tabela s√≥
otu_table &lt;- cbind(t1,t2,t3,t4,t5,t6,t7,t8,t9,t10,t11,t12,t13,t14,t15)
tax_table1 &lt;-as.data.frame(tax_table) # Converte a tax_table em um data.frame
row.names(otu_table) &lt;- tax_table1$Family # Coloca os nomes das linhas na nova otu_table criada baseado nos nomes das familias da tax_table
data.prop_AB &lt;- otu_table # troca de nome a otu_table
data.dist &lt;- vegdist(data.prop_AB, method = "bray") # Calcula as distancias entre as familias usando a m√©trica Bray-Curtis
row.clus &lt;- hclust(data.dist, "average")  # Faz a clusteriza√ß√£o usando a matriz de dist√¢ncias calculada anteriormente 
data.dist.g &lt;- vegdist(t(data.prop_AB), method = "bray") # Calcula as dist√¢ncias entre as amostras usando a m√©trica Bray-Curtis
col.clus &lt;- hclust(data.dist.g, "aver") # Faz a clusteriza√ß√£o usando a matriz de dist√¢ncias calculada anteriormente 
mat_data &lt;- data.matrix(data.prop_AB) # Transforma os dados a matrix
mat_data &lt;- round(mat_data, digits=1) # Coloca s√≥ uma casela decimal
Colors &lt;- rev(brewer.pal(5,"Spectral"))
Colors=colorRampPalette(Colors)(100) # Cria uma paleta de cores
```

---

### Graficando heatmap com clusteriza√ß√£o com "heatmap.2()"

.pull-left[

```r
# Cria o heatmap
heatmap_clustering &lt;- heatmap.2(mat_data, Rowv = as.dendrogram(row.clus), # Add o dendograma das linhas (familias)
          Colv = as.dendrogram(col.clus),  # Add o dendograma das colunas (amostras)
          dendrogram = "both",   # Indica que √© para graficar os dois dendogramas (linhas e colunas)
          col = Colors, # Add a paleta de cores 
          trace = "none", 
          lmat=rbind( c(0, 3, 4), c(2,1,0 ) ), lwid=c(1.5, 4, 2 ),
          margins = c(12, 9), density.info = "none",
          key.xlab="Percentage", key.ylab = NULL, key.title="", keysize=1.0, key = TRUE, # Personaliza a legenda
          cexRow=1.0, # Tamanho do texto no eixo y
          cexCol=1.0, # Tamanho do texto no eixo x
          srtCol=60,adjCol = c(1.0,0),# angulo e posi√ß√£o do texto no eixo x
          cellnote = mat_data, # Valores em cada c√©lula, no caso a porcentagem
          notecol="black",notecex=0.75, # Cor e tamanho do texto em cada c√©lula
          xlab = "Samples", main = "Heatmap" , ylab = "Families", # T√≠tulos dos eixos e do gr√°fico
          key.par = list(mar=c(3,3,0,3)), # Margens ("bottom.margin", "left.margin", "top.margin", "left.margin") )
          offsetRow=0, offsetCol=0) # Dist√¢ncias dos nomes nos eixos ao gr√°fico
```
]

.pull-right[
![](index_files/figure-html/plot-label-out29-1.png)&lt;!-- --&gt;
]

---
class: center, middle, inverse

# Conclus√µes

---
class: middle

* O pacote phyloseq tem uma s√©rie de fun√ß√µes que permitem a manipula√ß√£o integral de dados de sequenciamento massivo de marcadores filogen√©ticos

* O pacote **Phyloseq** em combina√ß√£o com outros pacotes, principalmente **ggplot2** proporcionam ferramentas para uma an√°lises integral de microbiomas.

* Com estes pacotes voc√™ pode gerar gr√°ficos *ready to print*, com qualidade e resolu√ß√£o √≥tima para publicar em revistas cientificas.

---
class: center, middle, inverse

.center[
![Saw](https://media.giphy.com/media/l4pTjOu0NsrLApt0Q/giphy.gif)
]
---

class: center, middle, inverse

# Gracias!
.center[
![Saw](https://media.giphy.com/media/3oriO5no5uDH8UGKbK/giphy.gif)
]


    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"navigation": {
"scroll": false
}
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
